app/ante/eth.go

```diff
@@ -424,11 +423,20 @@ func (vbd EthValidateBasicDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simu
 		baseFee := vbd.evmKeeper.GetBaseFee(ctx, ethCfg)
 
 		for _, msg := range protoTx.GetMsgs() {
+			if err := msg.ValidateBasic(); err != nil {
+				return ctx, sdkerrors.Wrap(err, "msg basic validation failed")
+			}
+
 			msgEthTx, ok := msg.(*evmtypes.MsgEthereumTx)
 			if !ok {
 				return ctx, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "invalid message type %T, expected %T", msg, (*evmtypes.MsgEthereumTx)(nil))
 			}
 
+			// Validate `From` field
+			if msgEthTx.From != "" {
+				return ctx, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "invalid From %s, expect empty string", msgEthTx.From)
+			}
```

Comment:
I thought this was checked on the message valiadation?

Reply:
message validation will accept non empty `From`, because it's used in places other than ante handler, but in ante handler we want more strict.